core/services/directrequestocr/drlistener.go

```diff
@@ -290,3 +291,34 @@ func (l *DRListener) markLogConsumed(lb log.Broadcast, qopts ...pg.QOpt) {
 func formatRequestId(requestId [32]byte) string {
 	return fmt.Sprintf("0x%x", requestId)
 }
+
+func (l *DRListener) timeoutRequests() {
+	defer l.shutdownWaitGroup.Done()
+	timeoutSec, freqSec, batchSize := l.pluginConfig.RequestTimeoutSec, l.pluginConfig.RequestTimeoutCheckFrequencySec, l.pluginConfig.RequestTimeoutBatchLookupSize
+	if timeoutSec == 0 || freqSec == 0 || batchSize == 0 {
+		l.logger.Warnw("request timeout checker not configured - disabling it")
+		return
+	}
+	ticker := time.NewTicker(time.Duration(freqSec) * time.Second)
+	for {
+		select {
+		case <-l.chStop:
+			ticker.Stop()
+			return
+		case <-ticker.C:
+			reqs, err := l.pluginORM.FindOldestEntriesByState(RESULT_READY, batchSize)
+			if err != nil {
+				l.logger.Errorw("error when calling FindOldestEntriesByState", "err", err)
+				break
+			}
+			for _, req := range reqs {
+				if req.ReceivedAt.Before(time.Now().Add(-(time.Duration(timeoutSec) * time.Second))) {
```

Comment:
Does not this deserve a dedicated ORM method that would return timed out entries?

Reply:
You're right. Current approach can be wasteful if there's nothing to be timed out (most common case). Thanks!